NMS-8861: Add information about logging

mfuhrmann opened the pull request:
### All Contributors

* [x] Have you read our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [x] Have you (electronically) signed the [OpenNMS Contributor Agreement](https://cla-assistant.io/OpenNMS/opennms)?

### Contribution Checklist

* Please [make an issue in the OpenNMS issue tracker](https://issues.opennms.org) if there isn't one already.<br />Once there is an issue, please:
  1. update the title of this PR to be in the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the JIRA link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
  4. once you've created this PR, please link to it in a comment in the JIRA issue
  Don't worry if this sounds like a lot, we can help you get things set up properly.
* If this is a new or updated feature, is there documentation for the new behavior?
* If this is new code, are there unit and/or integration tests?
* If this PR targets a `foundation-*` branch, does it try to avoid changing files in `$OPENNMS_HOME/etc/`?

### What's Next?

A PR should be assigned at least 2 reviewers.  If you know that someone would be a good person to review your code, feel free to add them.

If you need help making additions or changes to the documentation related to your changes, please let us know.

In any case, if anything is unclear or you want help getting your PR ready for merge, please don't hesitate to say something in the comments here,
or in [the #opennms-development chat channel](https://chat.opennms.com/opennms/channels/opennms-development).

Once reviewer(s) accept the PR and the branch passes continuous integration, the PR is eligible for merge.

At that time, if you have commit access (are an OpenNMS Group employee or a member of the OGP) you are welcome to merge the PR when you're ready.
Otherwise, a reviewer can merge it for you.

Thanks for taking time to contribute!

### External References

* JIRA (Issue Tracker): https://issues.opennms.org/browse/NMS-8861

mfuhrmann commented:
@Bonrob2 @mmahacek @indigo423 I've added here the fundamental information that a user would need to set a daemon to DEBUG to be able to troubleshoot problems.

It's not that much so I decided to have it on one page for now. Also, I am open to hear your suggestions for structuring this article.
Please let me know what you think about that.

The [explanation](https://github.com/OpenNMS/opennms/pull/4550/files#diff-751335e9dfd789c06554df4a9579963e11d8749cb4b5f4095148bf5d28383727R53) of where to find the code that needs to be changed is not the best, but how could we write it in a better way without explaining how XML file structures work?
In the end, it is a tree or path like:
![image](https://user-images.githubusercontent.com/4995222/163674537-da71a53b-290c-4cc3-89d5-4529e51f68d7.png)

--- 
I would not go deeper into this topic for now since we know that the logging component is not working optimally.
For example, the categorization of log entries is not optimal. We get so many WARN logs that something is "not working properly" thus `WARN` can't be treated seriously anymore. Maybe INFO would be better etc.

Or: If you have multiple hundreds of nodes monitored, you can set the poller or collector daemon to DEBUG anymore. I mean it is possible of course, but you won't be able to handle the file increase/rotation anymore (at least when you want to find logs by reading the files from a console session). I made the experience that a logging cluster like ELK or Greylog is required in such cases. But I don't think we want to offer this as a solution.

What I want to say is, that in my opinion, the logging definitions in all components should be re-worked and unified. Then it makes sense to write more helpful content about the logging.

Bonrob2 commented:
@mfuhrmann . I think this is sufficient for now, if as you mention, the logging component is not working optimally. 
I think the explanation of where to find the configuration is fine, as explaining XML tree structure would be too much. I might suggest some rewording of that section to make it flow a bit better. 

Do we need to explain they type of logging each log level provides, or will users understand the difference between trace, info, and warning? Is there anywhere in the UI that they can manage logs? If so, we should include that in this section as well.

Bonrob2 commented:
Note also that as of H30 we will be exposing the logs in the UI, so it would be good to add this information to this section of the documentation. (see NMS-14189)

mmahacek commented:
I'm thinking the reference:daemons pages should get an update to also list the associated log file.  I'd personally rather the logging page not have a example `ls` output, but instead change it to a table with one row per filename, with columns to link either to the associated Operations page and reference:daemon page.

mfuhrmann commented:
> Do we need to explain they type of logging each log level provides, or will users understand the difference between trace, info, and warning? Is there anywhere in the UI that they can manage logs? If so, we should include that in this section as well.

I think it will be worth it at some point. But I have no idea where exactly the differences are right now.

> Note also that as of H30 we will be exposing the logs in the UI, so it would be good to add this information to this section of the documentation. (see NMS-14189)

That sounds great! I am really looking forward to it!

> I'm thinking the reference:daemons pages should get an update to also list the associated log file. I'd personally rather the logging page not have a example `ls` output, but instead change it to a table with one row per filename, with columns to link either to the associated Operations page and reference:daemon page.

Let me prepare that. :+1: Even if the logging is not perfect right now, this structure makes sense for the future.

mfuhrmann commented:
@Bonrob2 @mmahacek I made the changes Marks suggested. Please let's have a look, before I will continue. The collected for example got an anchor link to the log file.

Still, I don't know exactly how I can test those changes before I push my changes to GH. Maybe you can give me a hint? Then I could test better and you would not have to request so many changes (or at least I hope that).

mmahacek commented:
I'd suggest combining the operation/admin/logging/introduction and change-logging pages into one page.  They're short enough that it can be one document.

mmahacek commented on docs/modules/reference/pages/logging/introduction.adoc:
```diff
@@ -0,0 +1,90 @@
+| bsmd.log
+|
+
+| [[ref-logging-collectd.log]] collectd.log 
```
Don't need an anchor defined here
```suggestion
| collectd.log 
```

mmahacek replied:
There shouldn't be anchors inside a table.  Plus it doesn't make sense to link to a row in the table.

mmahacek commented on docs/modules/reference/pages/daemons/introduction.adoc:
```diff
@@ -13,101 +13,134 @@ See the xref:operation:admin/daemon-config-files.adoc[system administration] sec
 |===
```
line 12 above should be something like `[cols="1,3,1"]` to account for the new columns added.

mfuhrmann replied:
fixed

mfuhrmann commented:
@Bonrob2 @mmahacek Added also the logging levels.

Bonrob2 commented on docs/modules/operation/pages/admin/logging/change-logging.adoc:
```diff
@@ -0,0 +1,23 @@
+
+== Change logging
+
+{page-component-title} uses the https://logging.apache.org/log4j/[Apache Log4j] framework for managing how the logs are being written, structured or rotated.
```
```suggestion
{page-component-title} uses the https://logging.apache.org/log4j/[Apache Log4j] framework to manage how the logs are written, structured, or rotated.
```

Bonrob2 commented on docs/modules/operation/pages/admin/logging/change-logging.adoc:
```diff
@@ -0,0 +1,23 @@
+
+For troubleshooting purposes `DEBUG` can be used. 
+By saving the changes in this file {page-component-title} needs a few seconds to apply the changes.
+The additional `DEBUG` log output is now also being written to the corresponding log file.
```
```suggestion
The additional DEBUG log output is now also being written to the corresponding log file.
```

Bonrob2 commented on docs/modules/operation/pages/admin/logging/change-logging.adoc:
```diff
@@ -0,0 +1,23 @@
+The additional `DEBUG` log output is now also being written to the corresponding log file.
+
+NOTE: Don't forget to set the logging level back to `WARN` after changing it to an higher severity.
+Depending on the scale {page-component-title} can generate a lot more logs which will require more disk space.
```
```suggestion
Depending on the scale, {page-component-title} can generate a lot more logs which will require more disk space.
```

Bonrob2 commented on docs/modules/operation/pages/admin/logging/introduction.adoc:
```diff
@@ -0,0 +1,12 @@
+[[ga-logging-introduction]]
+= Logging
+
+{page-component-title} stores logs into different files in the filesystem depending and named on the daemon that was acting on a process.
```
```suggestion
{page-component-title} stores logs in different files in the file system depending and named on the daemon that was acting on a process.
```

Bonrob2 commented on docs/modules/reference/pages/logging/introduction.adoc:
```diff
@@ -0,0 +1,90 @@
+|
+
+| notifd.log
+| Information about the notifications that were or were not send.
```
```suggestion
| Information about the notifications that were or were not sent.
```

Bonrob2 commented on docs/modules/reference/pages/logging/introduction.adoc:
```diff
@@ -0,0 +1,90 @@
+| All logging about the service availability monitors.
+
+| provisiond.log
+| The node provisioning processes are being logged here. This includes also service detection and policy results.
```
```suggestion
| The node provisioning processes are being logged here.
This also includes service detection and policy results.
```

Bonrob2 commented on docs/modules/operation/pages/admin/logging/introduction.adoc:
```diff
@@ -0,0 +1,35 @@
+[[ga-logging-introduction]]
+= Logging
+
+{page-component-title} stores logs into different files in the filesystem depending and named on the daemon that was acting on a process.
```
```suggestion
{page-component-title} stores logs in different files in the file system depending and named on the daemon that was acting on a process.
```

Bonrob2 commented on docs/modules/operation/pages/admin/logging/introduction.adoc:
```diff
@@ -0,0 +1,35 @@
+
+{page-component-title} stores logs into different files in the filesystem depending and named on the daemon that was acting on a process.
+
+For example: Logs for xref:operation:service-assurance/introduction.adoc#ga-service-assurance[service assurance checks] that are done by `pollerd` are stored in a file called `pollerd.log`.
```
```suggestion
For example, logs for xref:operation:service-assurance/introduction.adoc#ga-service-assurance[service assurance checks] that are done by `pollerd` are stored in the `pollerd.log` file.
```

Bonrob2 commented on docs/modules/operation/pages/admin/logging/introduction.adoc:
```diff
@@ -0,0 +1,35 @@
+
+The log files can be found in `/var/log/opennms/`.
+
+== List of log files
```
```suggestion

```

Bonrob2 commented on docs/modules/operation/pages/admin/logging/introduction.adoc:
```diff
@@ -0,0 +1,35 @@
+[[ga-change-logging]]
+== Change logging
+
+{page-component-title} uses the https://logging.apache.org/log4j/[Apache Log4j] framework for managing how the logs are being written, structured or rotated.
```
```suggestion
{page-component-title} uses the https://logging.apache.org/log4j/[Apache Log4j] framework to manage how the logs are written, structured, or rotated.
```

Bonrob2 commented on docs/modules/operation/pages/admin/logging/introduction.adoc:
```diff
@@ -0,0 +1,35 @@
+
+The corresponding configuration in this file can be found in the XML tree path
+
+*configuration/loggers/root/DynamicThresholdFilter/KeyValuePair*
```
```suggestion
`configuration/loggers/root/DynamicThresholdFilter/KeyValuePair`
```

Bonrob2 commented on docs/modules/operation/pages/admin/logging/introduction.adoc:
```diff
@@ -0,0 +1,35 @@
+
+*configuration/loggers/root/DynamicThresholdFilter/KeyValuePair*
+
+The default logging level is `WARN`.
```
```suggestion
The default logging level is WARN.
```

Bonrob2 commented on docs/modules/operation/pages/admin/logging/introduction.adoc:
```diff
@@ -0,0 +1,35 @@
+*configuration/loggers/root/DynamicThresholdFilter/KeyValuePair*
+
+The default logging level is `WARN`.
+But the levels can be set to `TRACE`, `INFO`, `DEBUG`, `WARN`, or `ERROR` depending on the need.
```
```suggestion
You can set the levels to TRACE, INFO, DEBUG, WARN, or ERROR depending on the need.
```

Bonrob2 commented on docs/modules/operation/pages/admin/logging/introduction.adoc:
```diff
@@ -0,0 +1,35 @@
+The default logging level is `WARN`.
+But the levels can be set to `TRACE`, `INFO`, `DEBUG`, `WARN`, or `ERROR` depending on the need.
+
+For troubleshooting purposes `DEBUG` can be used. 
```
```suggestion
For troubleshooting, use DEBUG. 
```

Bonrob2 commented on docs/modules/operation/pages/admin/logging/introduction.adoc:
```diff
@@ -0,0 +1,35 @@
+But the levels can be set to `TRACE`, `INFO`, `DEBUG`, `WARN`, or `ERROR` depending on the need.
+
+For troubleshooting purposes `DEBUG` can be used. 
+By saving the changes in this file {page-component-title} needs a few seconds to apply the changes.
```
```suggestion
After saving the changes in this file, {page-component-title} needs a few seconds to apply the changes.
```

Bonrob2 commented on docs/modules/operation/pages/admin/logging/introduction.adoc:
```diff
@@ -0,0 +1,35 @@
+
+For troubleshooting purposes `DEBUG` can be used. 
+By saving the changes in this file {page-component-title} needs a few seconds to apply the changes.
+The additional `DEBUG` log output is now also being written to the corresponding log file.
```
```suggestion
The additional DEBUG log output is now also being written to the corresponding log file.
```

Bonrob2 commented on docs/modules/operation/pages/admin/logging/introduction.adoc:
```diff
@@ -0,0 +1,35 @@
+By saving the changes in this file {page-component-title} needs a few seconds to apply the changes.
+The additional `DEBUG` log output is now also being written to the corresponding log file.
+
+NOTE: Don't forget to set the logging level back to `WARN` after changing it to an higher severity.
```
```suggestion
NOTE: Don't forget to set the logging level back to WARN after changing it to a higher severity.
```

Bonrob2 commented on docs/modules/operation/pages/admin/logging/introduction.adoc:
```diff
@@ -0,0 +1,35 @@
+The additional `DEBUG` log output is now also being written to the corresponding log file.
+
+NOTE: Don't forget to set the logging level back to `WARN` after changing it to an higher severity.
+Depending on the scale {page-component-title} can generate a lot more logs which will require more disk space.
```
```suggestion
Depending on the scale, {page-component-title} can generate a lot more logs, which requires more disk space.
```

Bonrob2 commented on docs/modules/operation/pages/admin/logging/logging-level.adoc:
```diff
@@ -0,0 +1,30 @@
+| Description
+
+| INFO
+| anything that is likely to be useful to the average user to determine the current general behavior/state of the system
```
```suggestion
| Anything likely to be useful to the average user to determine the current general behavior/state of the system.
```

Bonrob2 commented on docs/modules/operation/pages/admin/logging/logging-level.adoc:
```diff
@@ -0,0 +1,30 @@
+| anything that is likely to be useful to the average user to determine the current general behavior/state of the system
+
+| WARNING
+| messages relating to code that has a failure but it might be temporary or recoverable, or code that is misconfigured but can continue running in a regressed state
```
```suggestion
| Messages relating to code that has a failure that might be temporary or recoverable, or code that is misconfigured but can continue running in a regressed state.
```

Bonrob2 commented on docs/modules/operation/pages/admin/logging/logging-level.adoc:
```diff
@@ -0,0 +1,30 @@
+| messages relating to code that has a failure but it might be temporary or recoverable, or code that is misconfigured but can continue running in a regressed state
+
+| ERROR
+| only things that cause code to stop working and be unrecoverable
```
```suggestion
| Only things that cause code to stop working and be unrecoverable.
```

Bonrob2 commented on docs/modules/operation/pages/admin/logging/logging-level.adoc:
```diff
@@ -0,0 +1,30 @@
+| only things that cause code to stop working and be unrecoverable
+
+| DEBUG
+| additional info that can help diagnose issues
```
```suggestion
| Additional information that can help diagnose issues.
```

Bonrob2 commented on docs/modules/operation/pages/admin/logging/logging-level.adoc:
```diff
@@ -0,0 +1,30 @@
+| additional info that can help diagnose issues
+
+| TRACE
+| even more additional info that is often only useful to developers, or is very high-output and thus could cause performance issues by enabling
```
```suggestion
| Even more additional information that is often useful only to developers, or is very high output and thus could cause performance issues when enabled.
```

Bonrob2 commented on docs/modules/operation/pages/admin/logging/logging-level.adoc:
```diff
@@ -0,0 +1,30 @@
+[[ga-logging-level]]
+
+== Logging Level
```
```suggestion
= Logging Level
```

mmahacek commented on docs/modules/operation/pages/admin/logging/introduction.adoc:
```diff
@@ -0,0 +1,32 @@
+The additional DEBUG log output is now also being written to the corresponding log file.
+
+NOTE: Don't forget to set the logging level back to WARN after changing it to a higher severity.
+Depending on the scale, {page-component-title} can generate a lot more logs, which requires more disk space.
```
```suggestion
Depending on the scale, {page-component-title} can generate a lot of logs, which requires more disk space.

== Logging levels

.Service daemon logging level options
[options="header"]
[cols="1,8"]
|===
| Level
| Description

| INFO
| Anything likely to be useful to the average user to determine the current general behavior/state of the system.

| WARN
| Messages relating to code that has a failure that might be temporary or recoverable, or code that is misconfigured but can continue running in a regressed state.
This is the recommended default log level.

| ERROR
| Only things that cause code to stop working and be unrecoverable.

| DEBUG
| Additional information that can help diagnose issues.

| TRACE
| Even more additional information that is often useful only to developers.
This mode will output very detailed logs and can cause performance issues if left enabled.
Use sparingly as necessary.
|===
```

mmahacek reviewed (changes requested):
The Daemons Introduction page shouldn't include logging info, and there doesn't need to be a dedicated Logging page within the reference section.  Each of the individual daemon pages currently has a list of config files for the daemon. What would be helpful is an additional table with log files associated with the daemon.

Something like this added to referece/daemons/collectd.adoc
```
.Collectd log file overview
[options="header"]
[cols="2,3"]

|===
| File
| Description

| collectd.log
| Logs relating to collector server events.

| instrumentation.log
| Collection statistics used to populate the Instrumentation Log Reader in the web UI admin section.
|===
```

mfuhrmann commented:
> The Daemons Introduction page shouldn't include logging info

I will remove the columns for the logs here.

> and there doesn't need to be a dedicated Logging page within the reference section

I will remove the logging/introduction.adoc file.

> Each of the individual daemon pages currently has a list of config files for the daemon.
 What would be helpful is an additional table with log files associated with the daemon.

Got it. I could imagine we probably have to add some config files for missing daemons. I will check this.

mfuhrmann commented:
@mmahacek I've added some examples of what it could look like. 
As I said there are a lot not described daemons. I've just added the logging file section to those new files. 
I didn't remove logging/introduction.adoc for now since I can use it as a reference for what files we have. Would remove it at the end.

mmahacek commented:
Sounds good.  Those other daemons do need to get documented, and placeholders to start those others can get rolled up into this.
(Might want to check your branch to pull as I just pushed a commit to clear a merge conflict)

mfuhrmann commented:
The file `logging/introduction.adoc` now just contains a few log files. I am not absolutely sure where to add them.
Here is the file: https://github.com/OpenNMS/opennms/pull/4550/files#diff-bea4ef875f7f21d84f292ee308018191fde1af2fd0e2e6cc89192e6b283a9965
After clarification, we can remove `logging/introduction.adoc` from this PR.

Also, there are daemons that don't have log files (or they just get created when the daemon is enabled and used). For example actions.
The list of log files I am using here was taken from a fresh ONMS installation.

mmahacek commented:
@mfuhrmann Do you mind if I make some commits on this branch? I've got some info to fill in some blanks, and it would be easier to push directly than to go through suggestions as some are outside the editing range where GitHub will allow comments.  If you prefer not, I can send over the suggestions separately.

mfuhrmann commented:
You can always. And it's always appreciated. The final result is the most important 👍

Am 26. Mai 2022 01:08:04 MESZ schrieb mmahacek ***@***.***>:
***@***.*** Do you mind if I make some commits on this branch? I've got some info to fill in some blanks, and it would be easier to push directly than to go through suggestions as some are outside the editing range where GitHub will allow comments.  If you prefer not, I can send over the suggestions separately.
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/OpenNMS/opennms/pull/4550#issuecomment-1137931848
>You are receiving this because you were mentioned.
>
>Message ID: ***@***.***>
-- 
Diese Nachricht wurde von meinem Android-Gerät mit K-9 Mail gesendet.

Bonrob2 commented on docs/modules/reference/pages/daemons/daemon-config-files/statsd.adoc:
```diff
@@ -0,0 +1,15 @@
+[[ref-daemon-config-files-statsd]]
+= Statsd
+
```
Could we add an introductory sentence here?

mmahacek replied:
```suggestion

The Statsd daemon generates statistical TopN reports based on node data.

```

Bonrob2 commented on docs/modules/reference/pages/daemons/introduction.adoc:
```diff
@@ -20,52 +20,56 @@ See the xref:operation:admin/daemon-config-files.adoc[system administration] sec
 
-| xref:daemons/daemon-config-files/collectd.adoc[Collectd]
+| Collectd
 | Collect daemon is responsible for collecting data from nodes.
```
```suggestion
| Collect daemon collects data from nodes.
```

mmahacek replied:
```suggestion
| Collects data from nodes to persist performance metrics over time.
```

Bonrob2 reviewed (changes requested):
Looks really great. A few minor changes and questions. Do you think we close off on this one soon?

mmahacek commented on docs/modules/reference/pages/daemons/daemon-config-files/discovery.adoc:
```diff
@@ -15,3 +17,16 @@
+| Description
+
+| discovery.log
+| Network discovery scan jobs and their results are stored in this file.
```
```suggestion
| Logs relating to discovery scans.
Once the discovery scan is complete, Provisiond will handle adding the node to the database for monitoring.
```

mmahacek commented on docs/modules/reference/pages/daemons/daemon-config-files/reportd.adoc:
```diff
@@ -0,0 +1,15 @@
+[[ref-daemon-config-files-reportd]]
+= Reportd
+
```
```suggestion

The report daemon handles database report generation and scheduling.

```

mmahacek commented on docs/modules/reference/pages/daemons/introduction.adoc:
```diff
@@ -20,52 +20,56 @@ See the xref:operation:admin/daemon-config-files.adoc[system administration] sec
 | EnhancedLinkd
 | Enhanced link daemon (Enlinkd) discovers connections between nodes using data generated by various link discovery protocols and accessible via SNMP.
 
+
```
```suggestion
```

mmahacek commented on docs/modules/reference/pages/daemons/daemon-config-files/collectd.adoc:
```diff
@@ -103,3 +105,18 @@ Attributes are extracted from the HTTP responses using a regular expression.
+| Logs relating to collector server events.
+
+| instrumentation.log
+| Collection statistics used to populate the Instrumentation Log Reader in the web UI admin section.
```
Yep. https://issues.opennms.org/browse/NMS-10393

Bonrob2 replied:
Hmm. I could pick that up. How hard could it be? :-D

mmahacek replied:
famous last words. (honestly, probably not too difficult)

Bonrob2 replied:
Added a PR for the instrumentation reader (https://github.com/OpenNMS/opennms/pull/4788). Once it's merged, we can x-reference to it.

mmahacek replied:
```suggestion
| Collection statistics used to populate the xref:operation:admin/webui/log-reader.adoc[Instrumentation Log Reader] in the web UI admin section.
```

mmahacek commented on docs/modules/reference/pages/daemons/introduction.adoc:
```diff
@@ -83,29 +86,28 @@ See the xref:operation:admin/daemon-config-files.adoc[system administration] sec
+| Checks SNMP interface admin status. Separate from Pollerd's SnmpMonitor.
 
 | Statsd
 |
```
```suggestion
| Generates statistical reports from node data.
```

mmahacek commented on docs/modules/reference/pages/daemons/introduction.adoc:
```diff
@@ -83,29 +86,28 @@ See the xref:operation:admin/daemon-config-files.adoc[system administration] sec
+| Telemetryd
 | The Telemetry daemon collects sensor data sent from nodes to {page-component-title}.
 
 | TestLoadLibraries
 |
 
```
I think this is more internal. May be able to omit it. Not entirely sure.
```suggestion
```

Bonrob2 commented on docs/modules/reference/pages/daemons/introduction.adoc:
```diff
@@ -44,28 +46,29 @@ See the xref:operation:admin/daemon-config-files.adoc[system administration] sec
 
-| Manager
+| KarafStartupMonitor
 |
```
```suggestion
| Used in startup script to detect if Karaf started properly.
```

Bonrob2 commented on docs/modules/reference/pages/daemons/introduction.adoc:
```diff
@@ -44,28 +46,29 @@ See the xref:operation:admin/daemon-config-files.adoc[system administration] sec
+This can be helpful for devices that cannot be directly polled to check service state.
 
 | PerspectivePoller
 |
```
```suggestion
| Queries aggregated percentage availability for a given specific application and its services, as part of the xref:operation/application-perspective-monitoring/application-perspective-monitoring.adoc[application perspective monitoring].
```

mmahacek replied:
```suggestion
| Handles Application Perspective Monitoring availability calculations.
```

Bonrob2 commented on docs/modules/reference/pages/daemons/daemon-config-files/queued.adoc:
```diff
@@ -0,0 +1,15 @@
+[[ref-daemon-config-files-queued]]
+= Queued
+
```
```suggestion
Related to scheduling tasks to run in the future.
```

mmahacek commented on docs/modules/reference/pages/daemons/daemon-config-files/perspectivepollerd.adoc:
```diff
@@ -0,0 +1,17 @@
+[[ref-daemon-config-files-perspectivepoller]]
+= PerspectivePoller
+
+Handles xref:operation:application-perspective-monitoring/introduction.adoc[application perspective monitoring].
```
```suggestion
Queries aggregated percentage availability for a given specific application and its services, as part of the xref:operation/application-perspective-monitoring/application-perspective-monitoring.adoc[application perspective monitoring].
```

mmahacek commented on docs/modules/reference/pages/daemons/daemon-config-files/perspectivepollerd.adoc:
```diff
@@ -0,0 +1,17 @@
+[[ref-daemon-config-files-perspectivepoller]]
+= PerspectivePoller
+
+Queries aggregated percentage availability for a given specific application and its services, as part of the xref:operation/application-perspective-monitoring/application-perspective-monitoring.adoc[application perspective monitoring].
```
```suggestion
Queries aggregated percentage availability for a given specific application and its services, as part of the xref:operation/application-perspective-monitoring/introduction.adoc[application perspective monitoring].
```

mmahacek commented on docs/modules/reference/pages/daemons/daemon-config-files/perspectivepollerd.adoc:
```diff
@@ -0,0 +1,17 @@
+[[ref-daemon-config-files-perspectivepoller]]
+= PerspectivePoller
+
+Queries aggregated percentage availability for a given specific application and its services, as part of the xref:operation/application-perspective-monitoring/introduction.adoc[application perspective monitoring].
```
operation:application-perspective-monitoring/introduction.html

mmahacek replied:
```suggestion
Queries aggregated percentage availability for a given specific application and its services, as part of the xref:operation:application-perspective-monitoring/introduction.adoc[application perspective monitoring].
```
